### ISSUE TITLE
Add a way to export recipes as JSON

### ISSUE THREAD

**Issue by MikeAlhayek** [OrchardCMS/OrchardCore#15383]
Many times we export small recipes. We don't really want to download it as compressed then uncompress it.

It would be much easier if we add an option to export recipes as json.

**Comment by hishamco** [comment]
This might be useful if the recipe size is too big

**Comment by ns8482e** [comment]
The file download zip file includes recipe json content. It's  it possible to read the json from zip - it's just need to be updated in document.

**Comment by ns8482e** [comment]
Yes but extracting from zip is easy. Downloading zip file vs json could be confusing as both are not different.

Better option to improve existing file download option to compress the file or not

**Comment by MikeAlhayek** [comment]
@ns8482e check out the PR please. I named them compressed vs uncompressed so it is not confusing. 

The deployment UI is build for extension by adding alternative deployment target provider. So I extended based on the existing structure. 

![image](https://github.com/OrchardCMS/OrchardCore/assets/24724371/151212bc-4475-4d45-87dc-03cea8ba13f7)

**Comment by hishamco** [comment]
I'm afraid I have to disagree this will let every target be compressed / uncompressed, a good option is to add a checkbox, so when it is checked the file will be compressed regardless of its format JSON, XML .. etc

**Comment by MikeAlhayek** [comment]
First of all, deployment only support JSON format. We do not support XML or any other format. So complicating usage/design just because it not a good idea.

I do not mind the idea of adding a checkbox to compress or not. But this will require to change lots of our existing setup. We'll need to change how we render the target to use shapes, also add new templates for adding the checkbox.

I don't think we need to overhaul our setup just to allow adding a checkbox.

The current structure allow us to extent the UI by adding a new deployment source easily which is the approach I took.

**Comment by hishamco** [comment]
> First of all, deployment only support JSON format. We do not support XML or any other format. 

I know, perhaps some of the clients already have

> Better option to improve existing file download with option to compress the file or not

Checkbox could be a good option IMHO regardless of the code change

**Comment by MikeAlhayek** [comment]
The fact that OC supports only JSON recipes, we don't care about other formats. I don't think it's a valid argument.

> Checkbox could be a good option IMHO regardless of the code change

If you have the time, feel free to submit a PR converting deployment rendering to shapes and add the checkbox. My intention with the submitted PR to utilize the structure that already exists "path of least resistance".

Also, changing the structure could also mean possible breaking change people. 


Implementing a battleship when we only need a boat is an overkill ;)

**Comment by ns8482e** [comment]
> @ns8482e check out PR #15384 please. I named them compressed vs uncompressed so it is not confusing. 
> 
> 
> 
> The deployment UI is build for extension by adding alternative deployment target provider. So I extended based on the existing structure. 
> 
> 
> 
> ![image](https://github.com/OrchardCMS/OrchardCore/assets/24724371/151212bc-4475-4d45-87dc-03cea8ba13f7)
> 
> 

Instead of having two targets Can we have single target File Download with two action with drop-down button? Default will be compressed, second will be JSON?

**Comment by MikeAlhayek** [comment]
@ns8482e check out the my last comment 

https://github.com/OrchardCMS/OrchardCore/issues/15383#issuecomment-1962760066

I do not want to reconstruct the entire setup. Deployment sources are open for extension using cards

**Comment by ns8482e** [comment]
@MikeAlhayek  thanks IMHO only one target `File Download` is enough.  Here it seems the ask is different file formats - that can easily be achieved by providing different actions buttons that provides additional route data for file format to same controller action.

**Comment by MikeAlhayek** [comment]
@ns8482e okay. How about this?

![image](https://github.com/OrchardCMS/OrchardCore/assets/24724371/96cb7ec3-6c2f-4e82-8851-940227b5cdc1)

**Comment by hishamco** [comment]
Much better, that's why I prefered the checkbox to avoid adding a dropdown on each target

**Comment by ns8482e** [comment]
> @ns8482e okay. How about this?

Looks good now
